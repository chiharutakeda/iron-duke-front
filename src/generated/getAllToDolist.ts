/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: getAllToDolist
// ====================================================

export interface getAllToDolist_getAllToDolist {
  __typename: "TodoListEntity";
  id: string;
  /**
   * 苗字を返す
   */
  firstName: string;
  /**
   * 名前を返す
   */
  lastName: string;
  /**
   * やることリストを返す
   */
  todo: string;
  /**
   * 登録日を返す
   */
  createdAt: any;
}

export interface getAllToDolist {
  getAllToDolist: getAllToDolist_getAllToDolist[] | null;
}
