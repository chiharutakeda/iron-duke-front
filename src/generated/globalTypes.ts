/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export interface TodoListDeleteInputType {
  ID: number;
}

export interface TodoListInputType {
  firstName: string;
  lastName: string;
  todo: string;
}

export interface UserInfoInputType {
  email: string;
  password: string;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
