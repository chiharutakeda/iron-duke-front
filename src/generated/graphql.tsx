import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
const defaultOptions =  {}
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** The javascript `Date` as string. Type represents date and time as the ISO Date string. */
  DateTime: any;
};


export type LoginResponse = {
  __typename?: 'LoginResponse';
  accessToken: Scalars['String'];
};

export type Mutation = {
  __typename?: 'Mutation';
  RegistToDo?: Maybe<TodoListEntity>;
  DeleteToDo: TodoListEntity;
  Login?: Maybe<LoginResponse>;
  createUser?: Maybe<UserInfoEntity>;
};


export type MutationRegistToDoArgs = {
  todo: TodoListInputType;
};


export type MutationDeleteToDoArgs = {
  deleteTODO: TodoListDeleteInputType;
};


export type MutationLoginArgs = {
  LoginData: UserInfoInputType;
};


export type MutationCreateUserArgs = {
  UserData: UserInfoInputType;
};

export type Query = {
  __typename?: 'Query';
  sampleHello: Scalars['String'];
  getToDolist?: Maybe<TodoListEntity>;
  getAllToDolist?: Maybe<Array<TodoListEntity>>;
  getAllUser?: Maybe<Array<UserInfoEntity>>;
};

export type TodoListDeleteInputType = {
  ID: Scalars['Float'];
};

export type TodoListEntity = {
  __typename?: 'TodoListEntity';
  id: Scalars['ID'];
  /** 苗字を返す */
  firstName: Scalars['String'];
  /** 名前を返す */
  lastName: Scalars['String'];
  /** やることリストを返す */
  todo: Scalars['String'];
  /** 登録日を返す */
  createdAt: Scalars['DateTime'];
};

export type TodoListInputType = {
  firstName: Scalars['String'];
  lastName: Scalars['String'];
  todo: Scalars['String'];
};

export type UserInfoEntity = {
  __typename?: 'UserInfoEntity';
  id: Scalars['ID'];
  /** emailを返す */
  email: Scalars['String'];
  /** passwordを返す */
  password: Scalars['String'];
  /** 登録日を返す */
  createdAt: Scalars['DateTime'];
};

export type UserInfoInputType = {
  email: Scalars['String'];
  password: Scalars['String'];
};

export type DeleteToDoMutationVariables = Exact<{
  deleteTODO: TodoListDeleteInputType;
}>;


export type DeleteToDoMutation = (
  { __typename?: 'Mutation' }
  & { DeleteToDo: (
    { __typename?: 'TodoListEntity' }
    & Pick<TodoListEntity, 'id' | 'firstName' | 'lastName'>
  ) }
);

export type GetAllToDolistQueryVariables = Exact<{ [key: string]: never; }>;


export type GetAllToDolistQuery = (
  { __typename?: 'Query' }
  & { getAllToDolist?: Maybe<Array<(
    { __typename?: 'TodoListEntity' }
    & Pick<TodoListEntity, 'id' | 'firstName' | 'lastName' | 'todo' | 'createdAt'>
  )>> }
);

export type RegistToDoMutationVariables = Exact<{
  todo: TodoListInputType;
}>;


export type RegistToDoMutation = (
  { __typename?: 'Mutation' }
  & { RegistToDo?: Maybe<(
    { __typename?: 'TodoListEntity' }
    & Pick<TodoListEntity, 'id' | 'firstName' | 'lastName' | 'todo'>
  )> }
);


export const DeleteToDoDocument = gql`
    mutation DeleteToDO($deleteTODO: TodoListDeleteInputType!) {
  DeleteToDo(deleteTODO: $deleteTODO) {
    id
    firstName
    lastName
  }
}
    `;
export type DeleteToDoMutationFn = Apollo.MutationFunction<DeleteToDoMutation, DeleteToDoMutationVariables>;

/**
 * __useDeleteToDoMutation__
 *
 * To run a mutation, you first call `useDeleteToDoMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useDeleteToDoMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [deleteToDoMutation, { data, loading, error }] = useDeleteToDoMutation({
 *   variables: {
 *      deleteTODO: // value for 'deleteTODO'
 *   },
 * });
 */
export function useDeleteToDoMutation(baseOptions?: Apollo.MutationHookOptions<DeleteToDoMutation, DeleteToDoMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<DeleteToDoMutation, DeleteToDoMutationVariables>(DeleteToDoDocument, options);
      }
export type DeleteToDoMutationHookResult = ReturnType<typeof useDeleteToDoMutation>;
export type DeleteToDoMutationResult = Apollo.MutationResult<DeleteToDoMutation>;
export type DeleteToDoMutationOptions = Apollo.BaseMutationOptions<DeleteToDoMutation, DeleteToDoMutationVariables>;
export const GetAllToDolistDocument = gql`
    query getAllToDolist {
  getAllToDolist {
    id
    firstName
    lastName
    todo
    createdAt
  }
}
    `;

/**
 * __useGetAllToDolistQuery__
 *
 * To run a query within a React component, call `useGetAllToDolistQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetAllToDolistQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetAllToDolistQuery({
 *   variables: {
 *   },
 * });
 */
export function useGetAllToDolistQuery(baseOptions?: Apollo.QueryHookOptions<GetAllToDolistQuery, GetAllToDolistQueryVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useQuery<GetAllToDolistQuery, GetAllToDolistQueryVariables>(GetAllToDolistDocument, options);
      }
export function useGetAllToDolistLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<GetAllToDolistQuery, GetAllToDolistQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useLazyQuery<GetAllToDolistQuery, GetAllToDolistQueryVariables>(GetAllToDolistDocument, options);
        }
export type GetAllToDolistQueryHookResult = ReturnType<typeof useGetAllToDolistQuery>;
export type GetAllToDolistLazyQueryHookResult = ReturnType<typeof useGetAllToDolistLazyQuery>;
export type GetAllToDolistQueryResult = Apollo.QueryResult<GetAllToDolistQuery, GetAllToDolistQueryVariables>;
export const RegistToDoDocument = gql`
    mutation RegistToDo($todo: TodoListInputType!) {
  RegistToDo(todo: $todo) {
    id
    firstName
    lastName
    todo
  }
}
    `;
export type RegistToDoMutationFn = Apollo.MutationFunction<RegistToDoMutation, RegistToDoMutationVariables>;

/**
 * __useRegistToDoMutation__
 *
 * To run a mutation, you first call `useRegistToDoMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useRegistToDoMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [registToDoMutation, { data, loading, error }] = useRegistToDoMutation({
 *   variables: {
 *      todo: // value for 'todo'
 *   },
 * });
 */
export function useRegistToDoMutation(baseOptions?: Apollo.MutationHookOptions<RegistToDoMutation, RegistToDoMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<RegistToDoMutation, RegistToDoMutationVariables>(RegistToDoDocument, options);
      }
export type RegistToDoMutationHookResult = ReturnType<typeof useRegistToDoMutation>;
export type RegistToDoMutationResult = Apollo.MutationResult<RegistToDoMutation>;
export type RegistToDoMutationOptions = Apollo.BaseMutationOptions<RegistToDoMutation, RegistToDoMutationVariables>;