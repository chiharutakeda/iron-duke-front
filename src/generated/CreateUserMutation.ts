/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { UserInfoInputType } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: CreateUserMutation
// ====================================================

export interface CreateUserMutation_createUser {
  __typename: "UserInfoEntity";
  id: string;
  /**
   * emailを返す
   */
  email: string;
  /**
   * passwordを返す
   */
  password: string;
  /**
   * 登録日を返す
   */
  createdAt: any;
}

export interface CreateUserMutation {
  createUser: CreateUserMutation_createUser | null;
}

export interface CreateUserMutationVariables {
  createUserUserData: UserInfoInputType;
}
