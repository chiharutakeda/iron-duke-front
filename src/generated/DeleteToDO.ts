/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { TodoListDeleteInputType } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: DeleteToDO
// ====================================================

export interface DeleteToDO_DeleteToDo {
  __typename: "TodoListEntity";
  id: string;
  /**
   * 苗字を返す
   */
  firstName: string;
  /**
   * 名前を返す
   */
  lastName: string;
}

export interface DeleteToDO {
  DeleteToDo: DeleteToDO_DeleteToDo;
}

export interface DeleteToDOVariables {
  deleteTODO: TodoListDeleteInputType;
}
