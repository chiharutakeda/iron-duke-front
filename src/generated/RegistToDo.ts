/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { TodoListInputType } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: RegistToDo
// ====================================================

export interface RegistToDo_RegistToDo {
  __typename: "TodoListEntity";
  id: string;
  /**
   * 苗字を返す
   */
  firstName: string;
  /**
   * 名前を返す
   */
  lastName: string;
  /**
   * やることリストを返す
   */
  todo: string;
}

export interface RegistToDo {
  RegistToDo: RegistToDo_RegistToDo;
}

export interface RegistToDoVariables {
  todo: TodoListInputType;
}
