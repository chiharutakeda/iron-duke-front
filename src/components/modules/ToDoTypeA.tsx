import React from 'react';
import AtomButton from '@/components/atoms/AtomButton';
import { getAllToDolist_getAllToDolist } from '@/generated/getAllToDolist';
import { deleteToDolist } from '@/fetch/fetcher';
import styles from './styles/AllToDolist.module.css';

type ToDoTypeA = {
  todo: getAllToDolist_getAllToDolist;
};

const ToDoTypeA = ({ todo }: ToDoTypeA) => {
  return (
    <div className={styles.AllToDolist}>
      <p>
        <span>{'ID ' + todo.id}</span>
        <span>{'FIRSTNAME ' + todo.firstName}</span>
        <span>{'LASTNAME ' + todo.lastName}</span>
      </p>
      <p>{'TODO ' + todo.todo}</p>
      <AtomButton
        buttonType="DELETE"
        fnc={() => {
          deleteToDolist({ deleteTODO: { ID: Number(todo.id) } });
        }}
      />
    </div>
  );
};

export default ToDoTypeA;
