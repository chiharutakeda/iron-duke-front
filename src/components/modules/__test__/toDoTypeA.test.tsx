import ReactTestRenderer from 'react-test-renderer';
import ToDoTypeA from '../ToDoTypeA';
import {getAllToDolist_getAllToDolist} from '@/generated/getAllToDolist'

it('ToDotypeA', async () => {
  const todo: getAllToDolist_getAllToDolist = {
    __typename: 'TodoListEntity',
    createdAt: '2021-05-22T00:35:05.531Z',
    firstName: 'dddd',
    id: '61',
    lastName: 'fcf',
    todo: 'ddd',
  };
  const tree = await ReactTestRenderer.create(
    <ToDoTypeA todo={todo} key={4} />
  ).toJSON();
  await expect(tree).toMatchSnapshot();
});
