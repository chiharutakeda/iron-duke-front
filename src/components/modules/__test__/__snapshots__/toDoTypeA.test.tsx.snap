// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDotypeA 1`] = `
<div
  className="AllToDolist"
>
  <p>
    <span>
      ID 61
    </span>
    <span>
      FIRSTNAME dddd
    </span>
    <span>
      LASTNAME fcf
    </span>
  </p>
  <p
    className="ToDolist"
  >
    TODO ddd
  </p>
  <button
    className="DeleteButton"
    disabled={false}
    onClick={[Function]}
  >
    DELETE
  </button>
</div>
`;
